mobile/src/navigation/app/common/helpers/group-by-key/group-by-key.helper.ts

```diff
@@ -0,0 +1,18 @@
+import { DrawerNavigationItem } from '../../types/types';
+
+const groupByKey = (
+  items: DrawerNavigationItem[],
+  key: keyof DrawerNavigationItem,
+): DrawerNavigationItem[][] => {
+  return items.reduce((hash, item) => {
+    if (item[key] === undefined) return hash;
+
+    if (typeof key === 'string') {
+      return Object.assign(hash, {
+        [`${item[key]}`]: (hash[`${item[key]}`] || []).concat(item),
+      });
+    }
```

Comment:
`typeof key === 'string'` is unnecessary check, also we can make it more readable
```suggestion
    const groupName = item[key];

    if (typeof groupName !== 'string') {
      return hash;
    };
    
    const group = hash[groupName] ?? [];
    
    return {
      ...hash,
      [groupName]: group.concat(item),
    };
```